application-ideas-ui/src/main/resources/Ideas/IdeasClass.xml

```diff
@@ -508,121 +508,135 @@
-      status = button.siblings('.status').text();
+      docName = button.siblings('.fullName').data('reference');
+      status = button.siblings('.status').data('status');
     }
-
-    if(xm.document == 'Ideas.WebHome') {
-      var docTitle = button.parent().siblings('.doc_title');
+    // The vote is cast from the livetable of Ideas.Webhome
+    if (xm.document == 'Ideas.WebHome') {
       docName = button.parent().siblings('.doc_fullName').text();
     }
 
-    if(status != "$services.localization.render('Ideas.IdeasClass_status_open')") {
+    if (status != "$services.localization.render('Ideas.IdeasClass_status_open')") {
       new XWiki.widgets.Notification(
         "$services.localization.render('ideas.ideasclass.vote.closed')", 'warning');
       return;
     }
+    if (docName != '') {
+      var docRef = XWiki.Model.resolve(docName, XWiki.EntityType.DOCUMENT, [xm.wiki]);
+      url = new XWiki.Document(docRef).getRestURL('idea');
+    }
     var reflink = button.attr('class').split(' ')[1];
     var opp = button.data('opp');
     var refresh = button.data('ref');
-    // Creating the url according to the space from where the request was made
-    // and the type of action that is nedded.
-    if(voteClass == 'voted-pro') {
-      var addVote = 'addVote';
+    // Determine the vote type that was cast
+    if (voteClass == 'voted-pro') {
+      var voterGroup = 'supporters';
     } else {
-      var addVote = 'addVoteOpp';
+      var voterGroup = 'opponents';
     }
-
-    var url = new XWiki.Document('IdeasVoteService', 'Ideas').getURL('get',
-      ('xpage=plain&amp;outputSyntax=plain&amp;page=' + encodeURIComponent(docName) + '&amp;action=' + addVote));
-    // Ajax request.
     $.ajax({
-      url : url,
-      success : function(response) {
-        var voted = Array.from($('.' + reflink));
-        var opponent = Array.from($('.' + opp));
-        var supportText = $('.result-pro').next();
-        var againstText = $('.result-against').next();
-        // Changing the class acording to the vote. Updating the text that shows to the user which one is his choice.
-        if(response.result) {
-          voted.forEach(function(e) {
-            if($(e).hasClass(voteClass)) {
-              $(e).removeClass(voteClass);
-              supportText.empty();
-              againstText.empty();
-            } else {
-              $(e).addClass(voteClass);
-              if(voteClass == 'voted-pro') {
-                supportText.html("$services.localization.render('ideas.ideassheet.voted')");
+      headers: {
+        Accept: "application/json"
+      },
+      url: url,
+      type: 'GET',
+      success: function (idea) {
+        if (idea[voterGroup].includes(xm.userReference.toString())
+          || idea[voterGroup].includes(xm.userReference.relativeTo(xm.userReference.getRoot()).toString())) {
+          var action = 'DELETE';
+        } else {
+          var action = 'POST';
+        }
```

Comment:
Can't you determine this based on the UI state, in order to avoid making an additional GET request to get the server state?